tests/Mqtt5ClientCredentialsTest.cpp

```diff
@@ -0,0 +1,1185 @@
+    ASSERT_TRUE(mqtt5Client->Stop());
+
+    delete builder;
+    aws_string_destroy(endpoint);
+    aws_string_destroy(region);
+    aws_string_destroy(cognitoEndpoint);
+    aws_string_destroy(cognitoIdentity);
+    return AWS_OP_SUCCESS;
+}
+AWS_TEST_CASE(IoTMqtt5ConnectWSCognito, s_TestIoTMqtt5ConnectWSCognito)
+
+/*
+ * AWS Profile Provider connect for MQTT5
+ */
+static int s_TestIoTMqtt5ConnectWSProfile(Aws::Crt::Allocator *allocator, void *)
+{
+    struct aws_string *endpoint = NULL;
+    struct aws_string *region = NULL;
+    struct aws_string *profileCredentials = NULL;
+    struct aws_string *profileConfig = NULL;
+
+    int error = aws_get_environment_value(allocator, s_mqtt5_test_envName_iot_hostname, &endpoint);
+    error |= aws_get_environment_value(allocator, s_mqtt5_test_envName_iot_region, &region);
+    error |= aws_get_environment_value(allocator, s_mqtt5_test_envName_iot_profile_credentials, &profileCredentials);
+    error |= aws_get_environment_value(allocator, s_mqtt5_test_envName_iot_profile_config, &profileConfig);
+
+    bool isEveryEnvVarSet = (endpoint && region && profileCredentials && profileConfig);
+    if (isEveryEnvVarSet == true)
+    {
+        isEveryEnvVarSet =
+            (aws_string_is_valid(endpoint) && aws_string_is_valid(region) && aws_string_is_valid(profileCredentials) &&
+             aws_string_is_valid(profileConfig));
+    }
+    if (error != AWS_OP_SUCCESS || isEveryEnvVarSet == false)
+    {
+        printf("Environment Variables are not set for the test, skip the test");
+        aws_string_destroy(endpoint);
+        aws_string_destroy(region);
+        aws_string_destroy(profileCredentials);
+        aws_string_destroy(profileConfig);
+        return AWS_OP_SKIP;
+    }
+
+    ApiHandle apiHandle(allocator);
+
+    std::shared_ptr<Aws::Crt::Auth::ICredentialsProvider> provider = nullptr;
+    Aws::Crt::Auth::CredentialsProviderProfileConfig providerConfig;
+    providerConfig.ConfigFileNameOverride = aws_byte_cursor_from_c_str(aws_string_c_str(profileConfig));
+    providerConfig.CredentialsFileNameOverride = aws_byte_cursor_from_c_str(aws_string_c_str(profileCredentials));
+    provider = Aws::Crt::Auth::CredentialsProvider::CreateCredentialsProviderProfile(providerConfig);
+    if (!provider)
+    {
+        throw std::runtime_error("Failure to create credentials provider!");
+    }
+    Aws::Iot::WebsocketConfig websocketConfig(aws_string_c_str(region), provider);
+
+    Aws::Iot::Mqtt5ClientBuilder *builder = Aws::Iot::Mqtt5ClientBuilder::NewMqtt5ClientBuilderWithWebsocket(
+        aws_string_c_str(endpoint), websocketConfig, allocator);
+    ASSERT_TRUE(builder);
+
+    std::promise<bool> connectionPromise;
+    std::promise<void> stoppedPromise;
+
+    s_setupConnectionLifeCycle(builder, connectionPromise, stoppedPromise);
+
+    std::shared_ptr<Aws::Crt::Mqtt5::Mqtt5Client> mqtt5Client = builder->Build();
+
+    ASSERT_TRUE(mqtt5Client);
```

Comment:
Trivial: This 4 line block of ASSERT_TRUE is repeated 14 times and could potentially be a function that accepts the client and the promise?

There are also a number of tests, including this one, that don't wait on the stoppedPromise after the final ASSERT. Feel like we could probably add those since those same tests do assign them?